packages/blocks/src/matrix-question/index.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * External dependencies
+ */
+import { RawHTML } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import classnames from 'classnames';
+import { filter, first, isEmpty, join, map, times } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { useValidation } from '@crowdsignal/form';
+import { ErrorMessage, QuestionHeader, QuestionWrapper } from '../components';
+import Row from './row';
+
+/**
+ * Style dependencies
+ */
+import { MatrixCell, MatrixTable } from './styles';
+
+const MatrixQuestion = ( { attributes, className } ) => {
+	const errors = filter(
+		map( attributes.rows, ( row ) => {
+			// eslint-disable-next-line react-hooks/rules-of-hooks
+			const { error } = useValidation( {
+				fieldName: `q_${ attributes.clientId }[${ row.clientId }]${
+					attributes.multipleChoice ? '[]' : ''
+				}`,
+				validation: ( value ) => {
+					if ( attributes.mandatory && isEmpty( value ) ) {
+						return __( 'This question is required', 'blocks' );
+					}
+				},
+			} );
+
+			return error;
+		} )
+	);
+
+	const classes = classnames(
+		'crowdsignal-forms-matrix-question-block',
+		className,
+		{
+			'is-required': attributes.mandatory,
+			'is-error': ! isEmpty( errors ),
+		}
+	);
+
+	const tableStyles = {
+		gridTemplateColumns: join(
+			times( attributes.columns.length + 1, () => '1fr' ),
+			' '
+		),
+		gridTemplateRows: join(
+			times( attributes.rows.length + 1, () => '1fr' ),
+			' '
+		),
+	};
```

Comment:
Not for this PR, but I've seen that, often, we need to duplicate code on the `blocks-editor` and `blocks` packages.
It would be nice if we could think of a strategy to avoid these duplications.